### DISCUSSION TITLE
[bug]  TypeError: Cannot read properties of null (reading 'querySelector')

### DISCUSSION THREAD

**Discussion by vaynevayne** [ant-design/ant-design#44960] [General]
升级到最新版本出现的,
<img width="961" alt="image" src="https://github.com/ant-design/ant-design/assets/124666577/bd1a8fe7-f822-489a-bf1f-838ef51a0d3c">

<img width="944" alt="image" src="https://github.com/ant-design/ant-design/assets/124666577/3d392538-8d69-449c-82f0-2ba53a158e06">

**Comment by zombieJ** [comment]
搞个重现看看？

**Comment by vaynevayne** [comment]
我先确认下 我这里是否会存运行两个版本的 antd 

这个第三方包peerDependencies依赖5.5.x 的, 
我项目使用的是 5.9.x,
这种情况会下载两个版本吗? 

<img width="1003" alt="image" src="https://github.com/ant-design/ant-design/assets/124666577/9ad72702-408c-40f4-9a25-fce4afb8ae81">

**Reply by zombieJ** [reply]
peer 不会安装，应该会警告。

**Reply by vaynevayne** [reply]
pnpm@8 貌似会自动安装 peer 依赖, 但是我 node_moudles里就一个 antd,先不管了, 
不好复现, 就从代码本身找 bug吧

```jsx
export default function useContainerWidth(prefixCls) {
  const getContainerWidth = (ele, width) => {
    const container = ele.querySelector(`.${prefixCls}-container`);
    let returnWidth = width;
    if (container) {
      const style = getComputedStyle(container);
      const borderLeft = parseInt(style.borderLeftWidth, 10);
      const borderRight = parseInt(style.borderRightWidth, 10);
      returnWidth = width - borderLeft - borderRight;
    }
    return returnWidth;
  };
  return getContainerWidth;
}

  var onFullTableResize = function onFullTableResize(_ref3) {
    var width = _ref3.width;
    var mergedWidth = fullTableRef.current ? fullTableRef.current.offsetWidth : width;
    if (useInternalHooks && getContainerWidth && fullTableRef.current) { // 这里加个判断fullTableRef.current 存在, 就行了吧
// 看这里, mergedWidth 时判断了fullTableRef.current为空的情况, 而调用getContainerWidth函数时缺少判断

      mergedWidth = getContainerWidth(fullTableRef.current, mergedWidth) || mergedWidth;
    }
    if (mergedWidth !== componentWidth) {
      triggerOnScroll();
      setComponentWidth(mergedWidth);
    }
  };
```

**Reply by yoyo837** [reply]
`npm ls antd`看看便知

**Reply by vaynevayne** [reply]
嗯,  看到了就一个 5.9

**Reply by zombieJ** [reply]
`onFullTableResize` 来自 ResizeObserver，按理说不应该不存在。还是要给个重现看看为什么。

**Reply by vaynevayne** [reply]
那不知道了, 我这复现太难了, 我都是二次封装的 table

**Reply by zombieJ** [reply]
这么蛋疼？我比较担心是有更多的问题其实没有发现，草草兜底反而埋坑了

**Reply by vaynevayne** [reply]
带着错误裸奔, resize 也能正常工作, 等等其他人反馈吧,看有无线索

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/issues/40784 

看看是不是浏览器插件导致的。

**Comment by clown-Jack** [comment]
在切换路由的时候，报这个错Uncaught (in promise) TypeError: Cannot read properties of null (reading 'querySelector')
 "antd": "^5.9.2",
  "react": "^18.2.0",
如果把table的 virtual属性去掉，不会报这个错

**Reply by zombieJ** [reply]
能帮忙开个 issue 吗，需要有个重现看看~~~

**Comment by johnny-mh** [comment]
same issue here.
<img width="956" alt="image" src="https://github.com/ant-design/ant-design/assets/1061205/68a093ee-b1a9-425e-a3c3-481ad6e2b3f3">

<img width="545" alt="image" src="https://github.com/ant-design/ant-design/assets/1061205/b4b5f3fb-82fd-46ad-9c7a-5f3fecded586">

**Comment by INEWSK** [comment]
same issue

在Next.js dev 環境下，Table 組件下使用 scroll() ，在 Sider 進行頁面切換時會出現 TypeError: Cannot read properties of null (reading 'querySelector')

**Comment by adhemukhlis** [comment]
I got the same issue

if you just want to get rid of the warning, try downgrading `antd` to version `5.9.0`

```yarn add antd@5.9.0``` (without ^)

**Comment by qpeela** [comment]
It's fixed in rc-table v7.34.4 [url for commit](https://github.com/react-component/table/commit/65f9ce20ec71ee3a6b3eed23eaec8e027dc4f0a2)
